src/lasfilter_voxelgrid.cpp

```diff
@@ -0,0 +1,30 @@
+#include "lasfilter_voxelgrid.hpp"
+
+void voxelGrid::setVoxel(double vLen){
+    voxelSideLength = vLen;
+}
+
+int voxelGrid::getLength(double first, double last){
+    double distance = last - first;
+    int n = floor(distance / voxelSideLength);
```

Comment:
`int n`: considering that `last` is `double` and that `voxelSideLength` is `double` (likely to have a value close to 0.1 in most cases) I'm sure that `int` is not a safe type to store the output. Please check twice that you will never run into integer overflow. Check how LASlib manages that for `thin_with_grid`. If you think I'm wrong please explain.

Reply:
I agree that `int` is not the safest option here... to be safe I could change it to `long` or even `long long`, although it would use twice the memory :/

Reply:
Yep the issue is not the `int` actually the issue is the method chosen which is reliable only for TLS centered on 0 (or near) as you mentioned below.